src/core.js

```diff
@@ -1,144 +1,73 @@
+/*!
+ * QUnit @VERSION
+ * http://qunitjs.com/
+ *
+ * Copyright jQuery Foundation and other contributors
+ * Released under the MIT license
+ * http://jquery.org/license
+ *
+ * Date: @DATE
+ */
+
+import sourceFromStacktrace from "sourceFromStacktrace";
+import Assert from "assert";
+import Test from "test";
+import config from "config";
+
+// Only this module can modify these values
+/*globals QUnit:true, Date:true, setTimeout:true, clearTimeout:true*/
+
 var QUnit,
-	config,
 	onErrorFnPrev,
 	loggingCallbacks = {},
-	fileName = ( sourceFromStacktrace( 0 ) || "" ).replace( /(:\d+)+\)?/, "" ).replace( /.+\//, "" ),
 	toString = Object.prototype.toString,
 	hasOwn = Object.prototype.hasOwnProperty,
 	// Keep a local reference to Date (GH-283)
 	Date = window.Date,
```

Comment:
Does that only apply to `window`? What about `setTimeout` and other globals? How do you access the global scope correctly?